Q2.cpp

```diff
@@ -1,4 +1,4 @@
-Question1

```

Comment:
1) Syntax Error: Missing semicolon after the #include directive.
2) Syntax Error: Missing semicolon after the 'using namespace std;' statement.
3) Naming Convention Error: Class name 'subject' should start with an uppercase letter, e.g., 'Subject'.
4) Indentation Error: The code inside the class should be indented for better readability.
5) Commenting Error: There are no comments in the code to explain the purpose of the class or its methods.
6) Access Specifier Error: No access specifiers (public, private, or protected) are used in the class definition.
7) Code Organization Error: The code should be organized into separate files for class declaration (header file) and implementation (source file).